api/management-model/src/test/java/org/apache/polaris/core/admin/model/CatalogSerializationTest.java

```diff
@@ -0,0 +1,305 @@
+          original.getProperties().getDefaultBaseLocation(),
+          deserialized.getProperties().getDefaultBaseLocation(),
+          "Default base location should match");
+    } else {
+      assertNull(deserialized.getProperties(), "Properties should be null");
+    }
+
+    // Verify storage config
+    if (original.getStorageConfigInfo() != null) {
+      assertNotNull(deserialized.getStorageConfigInfo(), "Storage config should not be null");
+      assertEquals(
+          original.getStorageConfigInfo().getStorageType(),
+          deserialized.getStorageConfigInfo().getStorageType(),
+          "Storage type should match");
+
+      // Only check AWS-specific fields if we're explicitly using AWS storage config
+      if (original.getStorageConfigInfo() instanceof AwsStorageConfigInfo) {
+        assertTrue(
+            deserialized.getStorageConfigInfo() instanceof AwsStorageConfigInfo,
+            "Storage config should be AWS type");
+        AwsStorageConfigInfo originalAws = (AwsStorageConfigInfo) original.getStorageConfigInfo();
+        AwsStorageConfigInfo deserializedAws =
+            (AwsStorageConfigInfo) deserialized.getStorageConfigInfo();
+
+        assertEquals(
+            originalAws.getRoleArn(), deserializedAws.getRoleArn(), "Role ARN should match");
+      }
+
+      assertEquals(
+          original.getStorageConfigInfo().getAllowedLocations(),
+          deserialized.getStorageConfigInfo().getAllowedLocations(),
+          "Allowed locations should match");
+    } else {
+      assertNull(deserialized.getStorageConfigInfo(), "Storage config should be null");
+    }
+
+    return deserialized;
+  }
+
+  // Update testCatalogSerialization to use verifyRoundTrip
+  @Test
+  public void testCatalogSerialization() throws JsonProcessingException {
+    CatalogProperties properties = new CatalogProperties(TEST_LOCATION);
+
+    // Create AWS storage config with required roleArn
+    StorageConfigInfo storageConfig =
+        new AwsStorageConfigInfo(TEST_ROLE_ARN, StorageConfigInfo.StorageTypeEnum.S3);
+
+    Catalog catalog =
+        new Catalog(Catalog.TypeEnum.INTERNAL, TEST_CATALOG_NAME, properties, storageConfig);
+
+    verifyRoundTrip(catalog);
+  }
+
+  // Update testCatalogDeserialization
+  @Test
+  public void testCatalogDeserialization() throws JsonProcessingException {
+    StorageConfigInfo storageConfig =
+        new AwsStorageConfigInfo(TEST_ROLE_ARN, StorageConfigInfo.StorageTypeEnum.S3);
+    Catalog catalog =
+        new Catalog(
+            Catalog.TypeEnum.INTERNAL,
+            TEST_CATALOG_NAME,
+            new CatalogProperties(TEST_LOCATION),
+            storageConfig);
+    verifyRoundTrip(catalog);
+  }
+
+  // Update testCatalogWithNullFields - already using proper verification
+  @Test
+  public void testCatalogWithNullFields() throws JsonProcessingException {
+    Catalog catalog = new Catalog(Catalog.TypeEnum.INTERNAL, null, null, null);
+    verifyRoundTrip(catalog);
```

Comment:
I think having all those individual test methods that call into `verifyRoundTrip()` is confusing. Let's have one `@ParmeterizedTest` method with a `@MethodSource` to general all those input permutations.